packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -991,10 +992,17 @@ export const prepareVaultManagerKit = (
             AmountMath.add(state.totalCollateral, vault.getCollateralAmount()),
             oldCollateral,
           );
+
+          // add in new debt; subtract old debt
+          const debtPre = multiplyBy(
+            oldDebtNormalized,
+            state.compoundedInterest,
+          );
           state.totalDebt = AmountMath.subtract(
             AmountMath.add(state.totalDebt, vault.getCurrentDebt()),
-            oldDebtNormalized,
+            debtPre,
           );
```

Comment:
change looks good. consider inlining since it's not surprising that `oldDebtNormalized` needs to be actualized.
```js
          state.totalDebt = AmountMath.subtract(
            AmountMath.add(state.totalDebt, vault.getCurrentDebt()),
            multiplyBy(oldDebtNormalized, state.compoundedInterest)
          );
```